app/services/billable_metrics/aggregations/unique_count_service.rb

```diff
@@ -4,29 +4,41 @@ module BillableMetrics
   module Aggregations
     class UniqueCountService < BillableMetrics::Aggregations::BaseService
       def aggregate(from_datetime:, to_datetime:, options: {})
-        events = events_scope(from_datetime:, to_datetime:)
-          .where("#{sanitized_field_name} IS NOT NULL")
+        @from_datetime = from_datetime
+        @to_datetime = to_datetime
 
-        result.aggregation = events.count("DISTINCT (#{sanitized_field_name})")
-        result.pay_in_advance_aggregation = BigDecimal(compute_pay_in_advance_aggregation(events))
+        result.aggregation = compute_aggregation.ceil(5)
+        result.pay_in_advance_aggregation = BigDecimal(compute_pay_in_advance_aggregation)
         result.options = { running_total: running_total(options) }
-        result.count = events.count
+        result.count = result.aggregation
         result
       end
 
-      def compute_pay_in_advance_aggregation(events)
+      def compute_pay_in_advance_aggregation
         return 0 unless event
         return 0 if event.properties.blank?
 
-        existing_property = events
-          .where("#{sanitized_field_name} = '#{event.properties[billable_metric.field_name]}'")
-          .where.not(id: event.id)
-          .any?
+        unless previous_event
+          res = operation_type == :add ? 1 : 0
```

Comment:
Style/TernaryParentheses: Use parentheses for ternary expressions with complex conditions.